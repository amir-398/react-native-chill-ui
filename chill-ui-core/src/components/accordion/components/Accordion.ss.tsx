import type { AccordionPropsSs } from '@types';

import { BoxSs } from '@components/box';
import { PropsWithChildren } from 'react';

import styles from '../styles/Accordion.ss.styles';
import { AccordionProvider } from './AccordionContext';
import { accordionDefaultProps } from '../utils/defaultProps';

/**
 * Accordion component provides a collapsible content area with support for single or multiple items.
 *
 * @example
 * ```tsx
 * <Accordion type="single" collapsible>
 *   <AccordionItem value="item-1">
 *     <AccordionTrigger>Is it accessible?</AccordionTrigger>
 *     <AccordionContent>Yes. It adheres to the WAI-ARIA design pattern.</AccordionContent>
 *   </AccordionItem>
 * </Accordion>
 * ```
 *
 * @param children - Child AccordionItem components
 * @param type - Whether single or multiple items can be open at once ('single' | 'multiple')
 * @param collapseIcon - Icon to show when item is expanded (default: 'angle-down-solid')
 * @param collapsible - Whether items can be collapsed when type is 'single' (default: false)
 * @param defaultValue - Default open items (string for single, array for multiple)
 * @param disabled - Disable the entire accordion (default: false)
 * @param expandIcon - Icon to show when item is collapsed (default: 'angle-down-solid')
 * @param hasCollapseIcon - Whether to show collapse/expand icons (default: true)
 * @param iconPosition - Position of the expand/collapse icon (default: 'right')
 * @param onValueChange - Callback fired when accordion state changes
 * @param props - Additional ViewProps for the container
 *
 * @see {@link https://github.com/your-repo/chill-ui/tree/main/src/components/accordion/README.md Documentation}
 */
export default function Accordion(props: PropsWithChildren<AccordionPropsSs>) {
  const {
    children,
    collapseIcon = accordionDefaultProps.collapseIcon,
    collapsible = accordionDefaultProps.collapsible,
    defaultValue,
    disabled = accordionDefaultProps.disabled,
    expandIcon = accordionDefaultProps.expandIcon,
    hasCollapseIcon = accordionDefaultProps.hasCollapseIcon,
    iconPosition = accordionDefaultProps.iconPosition,
    onValueChange,
    style,
    type = accordionDefaultProps.type,
  } = props;

  return (
    <AccordionProvider
      type={type!}
      collapsible={collapsible}
      defaultValue={defaultValue}
      disabled={disabled}
      hasAnimation={false}
      animationDuration={0}
      hasCollapseIcon={hasCollapseIcon ?? true}
      iconPosition={iconPosition}
      expandIcon={expandIcon}
      collapseIcon={collapseIcon}
      onValueChange={onValueChange}
    >
      <BoxSs style={[styles.accordion, style]}>{children}</BoxSs>
    </AccordionProvider>
  );
}
