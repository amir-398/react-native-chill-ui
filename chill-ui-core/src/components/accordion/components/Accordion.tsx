import type { AccordionPropsTw } from '@types';

import { Box } from '@components/box';
import { PropsWithChildren } from 'react';
import { classNameHandler, cn, styleHandler } from '@utils';

import styles from '../styles/Accordion.styles';
import { AccordionProvider } from './AccordionContext';
import { accordionDefaultProps } from '../utils/defaultProps';
import { accordionClassName } from '../styles/Accordion.variants';
import { classNamePropsHandler } from '../../../utils/hybrid/classNamePropsHandler';

/**
 * Accordion component provides a collapsible content area with support for single or multiple items.
 * Automatically detects NativeWind availability and falls back to StyleSheet if needed.
 *
 * @example
 * ```tsx
 * <Accordion type="single" collapsible>
 *   <AccordionItem value="item-1">
 *     <AccordionTrigger>Is it accessible?</AccordionTrigger>
 *     <AccordionContent>Yes. It adheres to the WAI-ARIA design pattern.</AccordionContent>
 *   </AccordionItem>
 * </Accordion>
 * ```
 *
 * @param children - Child AccordionItem components
 * @param type - Whether single or multiple items can be open at once ('single' | 'multiple')
 * @param className - Custom CSS classes for styling
 * @param collapseIcon - Icon to show when item is expanded (default: 'angle-down-solid')
 * @param collapsible - Whether items can be collapsed when type is 'single' (default: false)
 * @param defaultValue - Default open items (string for single, array for multiple)
 * @param disabled - Disable the entire accordion (default: false)
 * @param expandIcon - Icon to show when item is collapsed (default: 'angle-down-solid')
 * @param hasCollapseIcon - Whether to show collapse/expand icons (default: true)
 * @param iconPosition - Position of the expand/collapse icon (default: 'right')
 * @param onValueChange - Callback fired when accordion state changes
 * @param props - Additional ViewProps for the container
 *
 * @see {@link https://github.com/your-repo/chill-ui/tree/main/src/components/accordion/README.md Documentation}
 */
export default function Accordion(props: PropsWithChildren<AccordionPropsTw>) {
  classNamePropsHandler(props, 'Accordion');
  const {
    children,
    className,
    collapseIcon = accordionDefaultProps.collapseIcon,
    collapsible = accordionDefaultProps.collapsible,
    defaultValue,
    disabled = accordionDefaultProps.disabled,
    expandIcon = accordionDefaultProps.expandIcon,
    hasCollapseIcon = accordionDefaultProps.hasCollapseIcon,
    iconPosition = accordionDefaultProps.iconPosition,
    onValueChange,
    style,
    type = accordionDefaultProps.type,
  } = props;

  return (
    <AccordionProvider
      type={type!}
      collapsible={collapsible}
      defaultValue={defaultValue}
      disabled={disabled}
      hasAnimation={false}
      animationDuration={0}
      hasCollapseIcon={hasCollapseIcon ?? true}
      iconPosition={iconPosition}
      expandIcon={expandIcon}
      collapseIcon={collapseIcon}
      onValueChange={onValueChange}
    >
      <Box
        {...classNameHandler(cn(accordionClassName, className))}
        {...styleHandler({ defaultStyle: styles.accordion, style })}
      >
        {children}
      </Box>
    </AccordionProvider>
  );
}
