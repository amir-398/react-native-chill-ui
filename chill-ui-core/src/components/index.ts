export * from './box';
export * from './toast';
export * from './loadingIndicatorsKit';
export * from './accordion';
export * from './dialog';
export * from './ripplePressable';
export * from './animatedBox';
export * from './icon';
export * from './avatar';
export * from './string';
export * from './inputDropdown';
export * from './AutocompleteDropdown';
export * from './highlightString';
export * from './buttonIcon';
export * from './checkbox';
export * from './chip';
export * from './maskedInput';
export * from './dropdownMenu';
export * from './inputSelectDropdown';
export * from './phoneNumberInput';
export * from './placesInput';
export * from './segmentedControl';
export * from './separator';
export * from './skeletons';
export * from './timePicker';
export * from './toggle';
export * from './wrapper';
