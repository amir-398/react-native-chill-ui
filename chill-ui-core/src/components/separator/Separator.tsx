import { cn } from '@utils';

import { Box } from '../box';
import { SeparatorProps } from '../../types/separator.types';

/**
 * Separator component that displays a horizontal line for visual separation.
 * A simple and lightweight component for creating visual dividers between content sections.
 *
 * @example
 * ```tsx
 * // Basic separator
 * <Separator />
 *
 * // Custom styled separator
 * <Separator className="h-0.5 bg-gray-300" />
 *
 * // Thick separator with custom color
 * <Separator className="h-1 bg-primary" />
 * ```
 *
 * @param className - Custom CSS classes for styling the separator
 */
export default function Separator({ className }: SeparatorProps) {
  return <Box className={cn('h-px bg-black', className)} />;
}
