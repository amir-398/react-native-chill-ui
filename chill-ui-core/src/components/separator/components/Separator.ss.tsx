import { BoxSs } from '@components/box';
import { SeparatorPropsSs } from '@types';

import { styles } from '../styles/Separator.ss.styles';

/**
 * Separator component that displays a horizontal line for visual separation.
 * A simple and lightweight component for creating visual dividers between content sections.
 *
 * @example
 * ```tsx
 * // Basic separator
 * <Separator />
 * ```
 *
 * @param style - Style object for the separator
 * @param viewProps - Rest of the view props
 */
export function Separator(props: SeparatorPropsSs) {
  const { style, ...rest } = props;
  return <BoxSs style={[styles.separator, style]} {...rest} />;
}

Separator.displayName = 'Separator';
