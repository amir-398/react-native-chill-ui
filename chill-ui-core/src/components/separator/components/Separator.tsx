import { Box } from '@components/box';
import { SeparatorPropsTw } from '@types';
import { classNameHandler, classNamePropsHandler, cn, styleHandler } from '@utils';

import { styles } from '../styles/Separator.ss.styles';
import { twStyles } from '../styles/Separator.tw.styles';

/**
 * Separator component that displays a horizontal line for visual separation.
 * A simple and lightweight component for creating visual dividers between content sections.
 * Automatically detects NativeWind availability and falls back to StyleSheet if needed.
 *
 * @example
 * ```tsx
 * // Basic separator
 * <Separator />
 * ```
 *
 * @param className - Custom CSS classes for styling the separator
 * @param style - Style object for the separator
 * @param viewProps - Rest of the view props
 */
export function Separator(props: SeparatorPropsTw) {
  classNamePropsHandler(props, 'Separator');
  const { className, style, ...rest } = props;
  return (
    <Box
      {...classNameHandler(cn(twStyles.separator, className))}
      {...styleHandler({ defaultStyle: styles.separator, style })}
      {...rest}
    />
  );
}

Separator.displayName = 'Separator';
