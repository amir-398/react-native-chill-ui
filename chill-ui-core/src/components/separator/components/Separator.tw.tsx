import { cn } from '@utils';
import { BoxTw } from '@components/box';
import { SeparatorPropsTw } from '@types';

import { twStyles } from '../styles/Separator.tw.styles';

/**
 * Separator component that displays a horizontal line for visual separation.
 * A simple and lightweight component for creating visual dividers between content sections.
 *
 * @example
 * ```tsx
 * // Basic separator
 * <Separator />
 * ```
 *
 * @param className - Custom CSS classes for styling the separator
 * @param viewProps - Rest of the view props
 */
export function Separator(props: SeparatorPropsTw) {
  const { className, ...rest } = props;
  return <BoxTw className={cn(twStyles.separator, className)} {...rest} />;
}

Separator.displayName = 'Separator';
