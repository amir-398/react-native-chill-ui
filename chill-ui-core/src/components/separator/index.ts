export { Separator } from './components/Separator';
export { Separator as SeparatorTw } from './components/Separator.tw';
export { Separator as SeparatorSs } from './components/Separator.ss';
