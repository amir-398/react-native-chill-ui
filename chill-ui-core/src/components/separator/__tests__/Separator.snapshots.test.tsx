import { render } from '@testing-library/react-native';

import { Separator } from '../components/Separator';

// Mock utils
jest.mock('../../../utils', () => ({
  classNameHandler: jest.fn(() => ({})),
  classNamePropsHandler: jest.fn(),
  cn: jest.fn((...args) => args.filter(Boolean).join(' ')),
  styleHandler: jest.fn(() => ({})),
}));

// Mock Box
jest.mock('../../../components/box', () => ({
  Box: ({ children, ...props }: any) => {
    const { View: ReactNativeView } = require('react-native');
    return (
      <ReactNativeView testID="box" {...props}>
        {children}
      </ReactNativeView>
    );
  },
}));

describe('Separator Snapshots', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot with basic props', () => {
    const tree = render(<Separator />);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom className', () => {
    const tree = render(<Separator className="h-0.5 bg-gray-300" />);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom style', () => {
    const tree = render(<Separator style={{ height: 2, backgroundColor: '#3B82F6' }} />);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with thick separator', () => {
    const tree = render(<Separator style={{ height: 4, backgroundColor: '#10B981' }} />);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom color', () => {
    const tree = render(<Separator style={{ backgroundColor: '#EF4444' }} />);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom styling', () => {
    const tree = render(
      <Separator
        style={{
          height: 2,
          backgroundColor: '#8B5CF6',
          marginVertical: 8,
          borderRadius: 1,
        }}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with accessibility props', () => {
    const tree = render(<Separator accessible accessibilityLabel="Content separator" testID="separator" />);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with all props combined', () => {
    const tree = render(
      <Separator
        className="custom-separator"
        style={{
          height: 2,
          backgroundColor: '#F59E0B',
          marginVertical: 12,
          borderRadius: 1,
          opacity: 0.8,
        }}
        accessible
        accessibilityLabel="Custom separator"
        testID="custom-separator"
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
