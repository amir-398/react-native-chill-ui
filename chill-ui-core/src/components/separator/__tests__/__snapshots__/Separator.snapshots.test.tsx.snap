// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator Snapshots should match snapshot with accessibility props 1`] = `
<View
  accessibilityLabel="Content separator"
  accessible={true}
  testID="separator"
/>
`;

exports[`Separator Snapshots should match snapshot with all props combined 1`] = `
<View
  accessibilityLabel="Custom separator"
  accessible={true}
  testID="custom-separator"
/>
`;

exports[`Separator Snapshots should match snapshot with basic props 1`] = `
<View
  testID="box"
/>
`;

exports[`Separator Snapshots should match snapshot with custom className 1`] = `
<View
  testID="box"
/>
`;

exports[`Separator Snapshots should match snapshot with custom color 1`] = `
<View
  testID="box"
/>
`;

exports[`Separator Snapshots should match snapshot with custom style 1`] = `
<View
  testID="box"
/>
`;

exports[`Separator Snapshots should match snapshot with custom styling 1`] = `
<View
  testID="box"
/>
`;

exports[`Separator Snapshots should match snapshot with thick separator 1`] = `
<View
  testID="box"
/>
`;
